chain-signatures/node/src/protocol/triple.rs

```diff
@@ -523,6 +503,7 @@ impl TripleManager {
                         // Protocol done, remove it from the ongoing pool.
                         self.ongoing.remove(id);
                         self.introduced.remove(id);
+                        self.gc.insert(*id, Instant::now());
```

Comment:
the issue with inserting it once it's done generating is that the GC expires it before it's ever gets used

Reply:
GC means we will not accept messages with this ID and will not restart a completed or failed protocol.
But now I'm reconsidering the whole approach. Maybe we need to deal with that on the messaging layer.

Reply:
Currently GC is more than just a filter for the messages though. It's used for more accurate error reporting when we take triples as well.

I'm in favor of making a dedicated message filter where we alert the message channel that a particular id has been completed